src/main/scala/beam/utils/NetworkHelper.scala

```diff
@@ -0,0 +1,67 @@
+package beam.utils
+import java.util
+
+import com.google.common.collect.Maps
+import com.typesafe.scalalogging.LazyLogging
+import javax.inject.Inject
+import org.apache.commons.collections4.MapUtils
+import org.matsim.api.core.v01.network.{Link, Network}
+
+import scala.collection.JavaConverters._
+
+case class LinkWithIndex(link: Link, index: Int)
+
+trait NetworkHelper {
+  def totalNumberOfLinks: Int
+  def getLinkWithIndexUnsafe(linkId: String): LinkWithIndex
+  def getLinkIdUnsafe(index: Int): String
+  def getLinkWithIndex(linkId: String): Option[LinkWithIndex]
+  def getLinkId(index: Int): Option[String]
+}
+
+class NetworkHelperImpl @Inject()(network: Network) extends NetworkHelper with LazyLogging {
```

Comment:
This is great, however I was under the impression that the links and their mappings were constantly mutating? If not then I can definitely see why this would save a lot of cycles.

Reply:
Yes, it's immutable.